src/main/java/core/basesyntax/SalaryInfo.java

```diff
@@ -40,7 +44,40 @@
      * София - 900</p>
      */
     public String getSalaryInfo(String[] names, String[] data, String dateFrom, String dateTo)
-            throws Exception {
-        return null;
+            throws IllegalDateParametersException {
+
+        StringBuilder builder = new StringBuilder();
+        builder.append("Отчёт за период ")
+                .append(dateFrom).append(" - ")
+                .append(dateTo).append("\n");
+
+        DateTimeFormatter formatter = DateTimeFormatter.ofPattern("dd.MM.yyyy");
+
+        LocalDate parsedDateFrom = LocalDate.parse(dateFrom, formatter);
+        LocalDate parsedDateTo = LocalDate.parse(dateTo, formatter);
+
+        if (parsedDateFrom.isAfter(parsedDateTo)) {
+            throw new IllegalDateParametersException("Wrong parameters");
+        }
```

Comment:
better move this checking at the start of method)

Reply:
cant put it on the very beginning because of parsing dates)
fixed.